src/FeedGenerator.php

```diff
@@ -101,9 +104,114 @@ function () {
 		// Set the store address as taxable location.
 		add_filter( 'woocommerce_customer_taxable_address', array( $this, 'set_store_address_as_taxable_location' ) );
 
-		// Timeout actions.
-		add_action( 'action_scheduler_unexpected_shutdown', array( $this, 'handle_action_timeout' ), 10, 2 );
-		add_action( 'action_scheduler_failed_action', array( $this, 'maybe_handle_error_on_timeout' ) );
+		// PHP shuts down execution for some reason.
+		add_action( 'action_scheduler_unexpected_shutdown', array( $this, 'handle_unexpected_shutdown' ), 10, 2 );
+		// Action got an exception thrown.
+		add_action( 'action_scheduler_failed_execution', array( $this, 'handle_failed_execution' ), 10, 2 );
+	}
+
+	/**
+	 * Returns feed generator actions group name.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_group_name(): string {
+		return PINTEREST_FOR_WOOCOMMERCE_PREFIX;
+	}
+
+	/**
+	 * Unexpected shutdown handler.
+	 *
+	 * @param int        $action_id - Action Scheduler action ID.
+	 * @param array|null $error - Error details.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return void
+	 */
+	public function handle_unexpected_shutdown( int $action_id, ?array $error ) {
+		if ( ! $error || ! $this->is_timeout_error( $error ) ) {
+			return;
+		}
+		$action = ActionScheduler::store()->fetch_action( $action_id );
+		$hook   = $action->get_hook();
+		$args   = $action->get_args();
+
+		// If not Pinterest Feed Generator action - ignore.
+		if ( $hook !== $this->get_action_full_name( self::CHAIN_BATCH ) ) {
+			return;
+		}
+
+		// Check if the action had failed before.
+		if ( $this->is_failure_rate_above_threshold( $hook, $args ) ) {
+			self::log(
+				sprintf(
+					// Translators: 1. Action Scheduler hook name.
+					__(
+						'Feed Generator `%s` Action reschedule threshold has been reached. Quit.',
+						'pinterest-for-woocommerce'
+					),
+					$hook
+				)
+			);
+			return;
+		}
```

Comment:
Hey, @JPry ! What do you think if we remove this. IMHO it does not make sense to have it since AS has its own retry mechanism.